app/controllers/comments_controller.rb

```diff
@@ -1,20 +1,22 @@
 class CommentsController < ApplicationController
   def create
-    @comment = current_user.comments.new(comment_params)
-    @post = Post.find(comment_params[:post_id])
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create(text: comment_parameters[:text], author_id: current_user.id, post_id: @post.id)
 
-    if @comment.save
-      flash[:notice] = 'Comment has been created successfully'
-      redirect_to user_post_path @post, notice: 'Comment has been created successfully.'
-    else
-      flash[:alert] = 'Adding a comment failed.'
-      render :new, status: :unprocessable_entity
+    respond_to do |format|
+      format.html do
+        if @comment.save
+          redirect_to user_post_path(@post.author.id, @post.id), notice: 'Comment created successfully'
+        else
+          redirect_to user_post_path(@post.author.id, @post.id), alert: 'Comment not created, try again!'
+        end
+      end
```

Comment:
- [ ] Kindly provide the view and route for this action for now you just only have the controller but there is no route and view for this action provided so far. Kindly do it.

- [ ] Also provide a link for the comment for creating a new comment in the `post#show` view.

Reply:
## Hi @andersonlebon and thank you for the feedback I really appreciate it but the comment belongs to the user and the post. if you can see in my route file i made a route for the comments inside the post. because it belongs to a post. also, i displayed all the comments in the post_controller. you will see `@comment = @post.comments.all. `you have to scroll down and you will see the link to create a post and each post has a form to create a comment and generate a like.

Reply:
Hello @vic778

Reply:
I am not sure I can see the link to create a new post 

![image](https://user-images.githubusercontent.com/65068771/163233664-7e69e17b-7794-4879-b3a7-c761e97e5081.png)

Reply:
Are you sure you're in the form branch?

Reply:
Yes sure 
![image](https://user-images.githubusercontent.com/65068771/163234500-273c1e78-1ed0-40ab-9afd-058dfaf3a28e.png)

Reply:
interesting. can we join the meeting for 3 minutes? i wanna check something

Reply:
okay, no problem

Reply:
please @andersonlebon the change i have made is it much with the request you asked me ?

Reply:
Kindly allow me to recheck, please.

Reply:
Fantastic everything is okay now.

![image](https://user-images.githubusercontent.com/65068771/163238427-cd9b9bcf-2d32-4701-aa2c-76018e5907e1.png)

![image](https://user-images.githubusercontent.com/65068771/163238747-cd18e78d-236e-4219-8cce-0983734f3e53.png)

- [ ] Let's just see what the next reviewer will say.

Reply:
Thank you for your feedback